### DISCUSSION TITLE
Instance Id is using a deprecated method

### DISCUSSION THREAD

**Discussion by ViktorSondergaard-s360** [invertase/react-native-firebase#5083] [General]
Firebase changed from Instance id to Firebase installation ID (FID). https://firebase.google.com/docs/projects/manage-installations#java_4. So the Instance ID module is using FirebaseInstanceId which is deprecated. So this should probably be upgraded to FirebaseInstallations. As I don't believe Instance ID is useful anymore? As Firebase is now using FID in Analytics

**Comment by mikehardy** [comment]
PRs happily merged :-)

It's my understanding that while we are using the deprecated APIs, under the covers it's all routing through the installations code now

It is something that "we" (invertase et al) want to get to but it is not a priority at the moment so there is no ETA, thus PRs would be welcome

**Reply by ViktorSondergaard-s360** [reply]
So you don't think there would be a difference using Instance ID and Firebase Installation ID? I'm trying to change the code locally and it seems like i got the same result.

**Reply by mikehardy** [reply]
that is correct: I do not think there will be a difference. As with all things that assertion needs evidence though - I'm happy to hear you are poking around at it locally and you appear to have evidence the assertion they are the same is accurate

**Reply by tomipaul** [reply]
Hey @mikehardy I noticed all the links leading to iid documentation has been removed, what is the recommended way to get the app instance id now? Thanks.

**Reply by mikehardy** [reply]
CHANGELOG always has breaking change notices where I document forward port path 

https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#breaking-changes
https://invertase.io/blog/react-native-firebase-versioning
